src/test/java/games/strategy/triplea/ui/mapdata/DefaultColorsTest.java

```diff
@@ -0,0 +1,28 @@
+package games.strategy.triplea.ui.mapdata;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.NoSuchElementException;
+import java.util.function.IntUnaryOperator;
+import java.util.stream.IntStream;
+
+import org.junit.jupiter.api.Test;
+
+public final class DefaultColorsTest {
+  private final DefaultColors defaultColors = new DefaultColors();
+
+  @Test
+  public void nextColor_ShouldReturnNextColorWhenColorsAvailable() {
+    assertThat(defaultColors.nextColor(), is(notNullValue()));
+  }
+
+  @Test
+  public void nextColor_ShouldThrowExceptionWhenNoColorsAvailable() {
+    assertThrows(
+        NoSuchElementException.class,
+        () -> IntStream.iterate(0, IntUnaryOperator.identity()).forEach(i -> defaultColors.nextColor()));
```

Comment:
Question: Assuming no exception is being thrown: Would this iteration go on forever?

Reply:
Indeed.  I originally had used a range of [1..1000) but thought that was too arbitrary.  I agree the infinite loop is a possible issue, so I'll just revert back to an arbitrary upper limit.  If someone adds additional colors that causes the test to fail because it no longer exhausts the available colors, they can just fix the test at the same time. :smile:

I suppose I could also just make `COLORS` `@VisibleForTesting` so the test would know exactly how many iterations to run.  Do you have a preference?

Reply:
I'd go with option 3 where we store the first value of the iteration and check if this value is being returned again.
If it does return the test should fail.
This way we don't need to know how much different colors there are + we check if there's at least 1 element.

Reply:
Well you already check the second thing so whatever

Reply:
Option 4 Use 255^3 as upper limit.
Therr can't be more colors than that in the current RGB scheme.